app/services/BalanceRequestCacheService.scala

```diff
@@ -179,4 +192,43 @@ class BalanceRequestCacheServiceImpl @Inject() (
       _        <- deferred.complete(response)
     } yield ()
   }
+
+  private def validateResponseMessage(
+    messageType: MessageType,
+    responseMessage: String
+  ): EitherT[IO, BalanceRequestError, Elem] =
+    EitherT
+      .fromEither[IO] {
+        validator.validate(messageType, responseMessage)
+      }
+      .leftMap { errors =>
+        BalanceRequestError.xmlValidationError(messageType, errors)
+      }
+
+  def updateBalanceRequest(
+    recipient: MessageIdentifier,
+    messageType: MessageType,
+    responseMessage: String
+  ): IO[Either[BalanceRequestError, Unit]] = {
+    val updateBalance = for {
+      validated <- validateResponseMessage(messageType, responseMessage)
+
+      response <- EitherT.fromEither[IO] {
+        parser.parseResponseMessage(messageType, validated)
+      }
```

Comment:
Note to self before I forget - I wanted to move this validation and parsing down into the `BalanceRequestService` and leave this service to deal only with updating the cache using the result of the `BalanceRequestService`

Reply:
I have done this in #11